compiler/tests-compiler/gh1007.ml

```diff
@@ -158,166 +158,162 @@ let ()  = M.myfun M.x
      var x$0 = x;
      for(;;){
       if(! x$0) return 0;
-      var
-       next = x$0[1],
-       rev_sort =
-         function(n, l){
-          if(2 === n){
-           if(l){
-            var match = l[2];
-            if(match){
-             var
-              tl = match[2],
-              x2 = match[1],
-              x1 = l[1],
-              s =
-                0 < caml_int_compare(x1, x2)
-                 ? [0, x1, [0, x2, 0]]
-                 : [0, x2, [0, x1, 0]];
-             return [0, s, tl];
-            }
-           }
-          }
-          else if(3 === n && l){
-           var _d_ = l[2];
-           if(_d_){
-            var match$2 = _d_[2];
-            if(match$2){
-             var
-              tl$1 = match$2[2],
-              x3 = match$2[1],
-              x2$0 = _d_[1],
-              x1$0 = l[1],
-              s$0 =
-                0 < caml_int_compare(x1$0, x2$0)
-                 ? 0
-                   < caml_int_compare(x2$0, x3)
-                   ? [0, x1$0, [0, x2$0, [0, x3, 0]]]
-                   : 0
-                     < caml_int_compare(x1$0, x3)
-                     ? [0, x1$0, [0, x3, [0, x2$0, 0]]]
-                     : [0, x3, [0, x1$0, [0, x2$0, 0]]]
-                 : 0
-                   < caml_int_compare(x1$0, x3)
-                   ? [0, x2$0, [0, x1$0, [0, x3, 0]]]
-                   : 0
-                     < caml_int_compare(x2$0, x3)
-                     ? [0, x2$0, [0, x3, [0, x1$0, 0]]]
-                     : [0, x3, [0, x2$0, [0, x1$0, 0]]];
-             return [0, s$0, tl$1];
-            }
-           }
-          }
+      var next = x$0[1];
+      function sort(n, l){
```

Comment:
It seems that the function order got reversed, which makes thinks harder to compare.

Reply:
The ordering is altered by compiler/lib/strongly_connected_components.ml.
It should fix it both on master and in this PR.

Reply:
now fixed